### ISSUE TITLE
Ensure compatibility with runtime js on geodata

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#71]
The [score](https://pub.dev/packages/geodata/versions/0.8.0-a.7/score) for the pre-release `0.8.0-a.7` was *awarded with 120 pub points (of a possible 130)*.

The reason as reported at *support multiple platforms* section:
```
10/20 points: Supports 1 of 2 possible platforms (native, js)
Consider supporting multiple platforms:

Package not compatible with runtime js
Because:

package:geodata/geodata.dart that imports:
package:geodata/ogcapi_features_client.dart that imports:
package:geodata/src/ogcapi_features/service/client.dart that imports:
package:geodata/src/ogcapi_features/service/client/ogc_feature_client.dart that imports:
package:geodata/src/utils/features.dart that imports:
package:geodata/src/utils/features/feature_file_adapter.dart that imports:
dart:io
```

The `feature_file_adapter.dart ` class:

```dart
import 'dart:convert';
import 'dart:io' show File;

import '/src/core/features.dart';

/// Maps a JSON Object read from [location] to an entity using [toEntity].
Future<T> readEntityFromJsonObject<T>(
  File location, {
  required T Function(Map<String, Object?> data) toEntity,
}) async {
  try {
    // ensure a file exists
    if (!location.existsSync()) {
      throw const FeatureException(FeatureFailure.notFound);
    }

    // read file contents as text
    final text = await location.readAsString();

    // decode JSON and expect it to contain JSON Object as `Map<String, Object?`
    final data = json.decode(text) as Map<String, Object?>;

    // map JSON Object to an entity
    return toEntity(data);
  } on FeatureException {
    rethrow;
  } catch (e, st) {
    // other exceptions (including errors)
    throw FeatureException(FeatureFailure.clientError, cause: e, trace: st);
  }
}
```

This is currently used only by the `geojson_client` that provides http and file access. 

The warning is because of importing `dart:io` which is not supported on the web platform.

Solution choices
1. Use [universal_io](https://pub.dev/packages/universal_io), that supports web, to access files
2. Abstract `feature_file_adapter.dart` implementation so that dependency to `dart:io` (and `File`) must be injected by a client using the `geodata` package
3. Remove support for file data source on the `geojson_client`, and code a separate **geodata_io** (or similar) package that would introduce dependencies to `dart:io` and maybe other server side capabilities too that are not present on web

**Comment by navispatial** [comment]
Maybe choice 3 might happen in some distant future. 

Choice 1 would be easy, but introducing a new dependency only for quite limited feature - acting as http / web client is the main task.

So go with choice 2.

**Comment by navispatial** [comment]
Adapter changed to this:

```dart
import 'dart:convert';

import '/src/core/features.dart';

/// Maps a JSON Object read from [source] to an entity using [toEntity].
///
/// The source function returns a future that fetches data from a file, a web
/// resource or other sources. Contents must be GeoJSON compliant data.
Future<T> readEntityFromJsonObject<T>(
  Future<String> Function() source, {
  required T Function(Map<String, Object?> data) toEntity,
}) async {
  try {
    // read contents as text
    final text = await source();

    // decode JSON and expect it to contain JSON Object as `Map<String, Object?`
    final data = json.decode(text) as Map<String, Object?>;

    // map JSON Object to an entity
    return toEntity(data);
  } on FeatureException {
    rethrow;
  } catch (e, st) {
    // other exceptions (including errors)
    throw FeatureException(FeatureFailure.clientError, cause: e, trace: st);
  }
}
```

**Comment by navispatial** [comment]
Implemented in pre-release `0.8.0-a.8` of the upcoming BETA-version `0.8.0`
